lib/get-format.js

```diff
@@ -36,7 +38,11 @@ function mimeToFormat(mime) {
 }
 
 /**
- * @type {Record<string, (parsed: URL) => string|null>}
+ * @typedef {(parsed: URL, context: {parentURL: string}, ignoreErrors: boolean) => string|null} ProtocolHandler
```

Comment:
You can use the longer form to more explicitly define functions:

```suggestion
 * @callback ProtocolHandler
 * @param {URL} parsed
 * @param {{parentURL: string}} context
 * @param {boolean} ignoreErrors
 * @returns {string|null}
```